src/Datadog.Trace.ClrProfiler.Managed/IntegrationVersionRange.cs

```diff
@@ -0,0 +1,92 @@
+namespace Datadog.Trace.ClrProfiler
+{
+    /// <summary>
+    /// Specifies a safe version range for an integration.
+    /// </summary>
+    public class IntegrationVersionRange
```

Comment:
Any reason we're not using the built-in [`System.Version`](https://docs.microsoft.com/en-us/dotnet/api/system.version) class?

Reply:
Granular control over the types was my thought. I wanted to make sure everything was clearly ushort up and down.
What advantage would System.Version give us?

Reply:
Not having to re-implement the conversions to and from string is what I was thinking. `Version` also has equality and relational operators, but I don't think we need those in C# code.

Remind me, why is using ushort so important here?

Reply:
It's not a huge deal, it's just that they're ultimately retrieved that way.
![image](https://user-images.githubusercontent.com/1801443/57331115-d97a3f80-70e5-11e9-8341-445569045487.png)

Reply:
And it prevents any shenanigans with overflows when loading them through the nlohmann serializer.
I could also make them int throughout, but ushort seemed very clear. Minimum v is 0, maximum v is the same maximum as the ASSEMBLYMETADATA we compare againt.

Reply:
That makes sense, thanks.